fix: prevent sparse spacing in hyphenated words

monotykamary opened the pull request:
## Summary

This PR fixes the sparse spacing issue that occurs when displaying hyphenated words like 'claude-code-action' in both Assistant and User messages in the TUI. Fixes #621.

## Problem

Previously, hyphenated words would break at hyphens, creating sparse single-word lines that made the chat interface look broken and difficult to read.

Before:
<img width="704" height="651" alt="image" src="https://github.com/user-attachments/assets/0aaf5435-5a2b-4ac3-9e2e-a47df65ba872" />


## Solution

After:
<img width="703" height="651" alt="image" src="https://github.com/user-attachments/assets/00cc4ae0-dbaa-40cb-a376-7f239987b626" />

- **Assistant messages**: Use non-breaking hyphens (U+2011) in markdown rendering to prevent word breaks at hyphens
- **User messages**: Apply non-breaking hyphens before word wrapping and styling
- **Textarea component**: Modified word boundary detection to treat hyphens as part of words
- **Width constraint**: Maintained 86 character width limit with precise calculations

## Changes Made

1. **packages/tui/internal/components/chat/message.go**: Updated user message rendering to use non-breaking hyphens
2. **packages/tui/internal/components/textarea/textarea.go**: Modified word boundary detection in textarea word wrapping
3. **packages/tui/internal/util/file.go**: Enhanced markdown rendering with hyphen handling

## Testing

- Verified both Assistant and User messages display correctly
- Confirmed 86 character width constraint is maintained
- Tested with various hyphenated words and phrases
- No visual regressions in existing functionality

This change significantly improves the readability of hyphenated content in the TUI without affecting any other functionality.

monotykamary committed:
fix: prevent sparse spacing in hyphenated words

- Use non-breaking hyphens (U+2011) to prevent word breaks at hyphens
- Apply fix to both Assistant (markdown) and User (chat) messages
- Maintain 86 character width constraint
- Fix word boundary detection in textarea component

Fixes sparse single-word lines when displaying hyphenated words like 'claude-code-action'

adamdotdevin commented on packages/tui/internal/components/chat/message.go:
```diff
@@ -196,17 +196,26 @@ func renderText(
 		base := styles.NewStyle().Foreground(t.Text()).Background(backgroundColor)
-		words := strings.Fields(text)
+		// Apply non-breaking hyphens to prevent sparse breaks
+		textWithHyphens := strings.ReplaceAll(text, "-", "2011")
```
is this supposed to be `u2011`?

monotykamary replied:
ah yes, my bad; let me add that back in

adamdotdevin commented on packages/tui/internal/components/chat/message.go:
```diff
@@ -196,17 +196,26 @@ func renderText(
+		wrappedText := ansi.WordwrapWc(styledText, width-6, " ")
+		
+		// Restore regular hyphens
+		finalText := strings.ReplaceAll(wrappedText, "2011", "-")
```
again, `u2011`?

adamdotdevin commented on packages/tui/internal/util/file.go:
```diff
@@ -83,11 +83,24 @@ func Extension(path string) string {
 
 func ToMarkdown(content string, width int, backgroundColor compat.AdaptiveColor) string {
-	r := styles.GetMarkdownRenderer(width-6, backgroundColor)
+	// Use precise width calculation to maintain 86 constraint
```
what is `86 constraint`?

adamdotdevin commented on packages/tui/internal/util/file.go:
```diff
@@ -83,11 +83,24 @@ func Extension(path string) string {
+	renderWidth := width - 6 // Account for padding and borders
+	if renderWidth < 10 {
+		renderWidth = 10
+	}
```
what's this for?

adamdotdevin reviewed (changes requested):
just a few questions, i'd love to have this fixed, just need to understand the implementation a little better

monotykamary committed:
fix: add missing u's

monotykamary commented:
Now that you've pointed it out, I think there might be a more ergonomic way to do this 🤔 
also now that we're going for a stretched look

monotykamary committed:
Merge branch 'dev' into fix/hyphen-word-wrapping

monotykamary committed:
refactor: make hyphen handling ergonomic and dynamic

- Extract hyphen preservation logic into reusable utility functions
- Replace magic numbers with dynamic style-based calculations
- Add GetMessageContainerFrame() and GetMarkdownContainerFrame() for DRY width calculations
- Simplify width calculations by removing outdated constraints
- Improve code maintainability with semantic function names and clear documentation

The hyphen handling now automatically adapts to style changes without manual updates.

monotykamary commented:
| Before | After |
| --- | ---|
| <img width="1401" height="497" alt="image" src="https://github.com/user-attachments/assets/e273bffd-efcf-4d36-8fdd-fcf1c0ea2986" /> | <img width="1401" height="421" alt="image" src="https://github.com/user-attachments/assets/89e4217f-34d4-4aa9-a684-8db5b736ee8a" /> |
